admin.go

```diff
@@ -335,6 +335,8 @@ func replaceLocalAdminServer(cfg *Config) error {
 		return err
 	}
 
+	serverMu.Lock()
+	defer serverMu.Unlock()
```

Comment:
The lock is only needed for L340, so we don't need a defer here and can move this unlock to L348.

Reply:
Makes sense, updated in f7b1ae558b11b7f61a2dcc6b9ec19f19d57c8265